sql/sql_acl.cc

```diff
@@ -8651,110 +8652,134 @@ bool mysql_grant_role(THD *thd, List <LEX_USER> &list, bool revoke)
   DBUG_RETURN(result);
 }
 
-
-static bool mysql_grant(THD *thd, const char *db, List<LEX_USER> &list,
-                        const Priv_spec &priv_spec, bool is_proxy)
+static bool mysql_grant_global(THD *thd,
+                               Grant_tables &tables,
+                               List<LEX_USER> &list,
+                               const Priv_spec &priv_spec,
+                               bool create_new_users,
+                               bool no_auto_create_users)
 {
-  List_iterator <LEX_USER> str_list (list);
-  LEX_USER *Str, *tmp_Str, *proxied_user= NULL;
-  char tmp_db[SAFE_NAME_LEN+1];
-  bool create_new_users=0;
-  int result;
-  DBUG_ENTER("mysql_grant");
+  List_iterator<LEX_USER> it(list);
+  bool result= false;
+  DBUG_ENTER("mysql_grant_global");
 
-  if (lower_case_table_names && db)
+  /* go through users in user_list */
+  for (LEX_USER *user= it++; user != NULL; user= it++)
   {
-    char *end= strnmov(tmp_db,db, sizeof(tmp_db));
-    if (end >= tmp_db + sizeof(tmp_db))
+    /* Resolve if it the proxy is "CURRENT_USER" or "CURRENT_ROLE" or
+       a username / rolename without a hostname specified. */
+    LEX_USER *resolved_user= get_current_user(thd, user, false);
+    if (!resolved_user)
     {
-      my_error(ER_WRONG_DB_NAME ,MYF(0), db);
-      DBUG_RETURN(TRUE);
+      result= true;
+      continue;
     }
-    my_casedn_str(files_charset_info, tmp_db);
-    db=tmp_db;
+    /* TODO(cvicentiu) This might actually not be necessary for grant_proxy.
+       Carry over auth information. */
```

Comment:
this comment is in `mysql_grant_global`, not in `mysql_grant_proxy`. looks out of place